### DISCUSSION TITLE
Streamlink options in python

### DISCUSSION THREAD

**Discussion by NiceShyGuy** [streamlink/streamlink#5173] [Q&A] [answered]
### Description

I'm trying to figure out how to setup options for streamlink and the mpv player properly. This is my current code:

```
options = {
    "http-headers": f"OAuth {environ['TWITCH_ACCESS_TOKEN']}",
    "hls-live-edge": 1,
    "hls-segment-stream-data": True,
    "stream-segment-threads": 2,
    "hls-playlist-reload-time": "live-edge", # is this valid?
}
session = streamlink.Streamlink(options=options)
```

I'm a bit confused if `hls-playlist-reload-time` is actually being set. Looking in the API guide for set_option it is not listed: https://streamlink.github.io/api.html#streamlink.Streamlink.set_option

But it is listed in the CLI:
https://streamlink.github.io/cli.html#cmdoption-hls-playlist-reload-time

Does this work? is it possible to set mpv `--volume` through these options? or should i just send commands directly to mpv to control things like this and other settings like `--no-cache`?

Thanks.

**Comment by bastimeyer** [comment, marked as answer]
> hls-playlist-reload-time

Looks like it's missing from the docstring in `Streamlink.set_option`, which needs to get fixed:
https://github.com/streamlink/streamlink/blob/1d525705a0506327a5d518bfac04271cee74f875/src/streamlink/session.py#L208
https://github.com/streamlink/streamlink/blob/1d525705a0506327a5d518bfac04271cee74f875/src/streamlink/session.py#L233-L355

Here's the mapping for all CLI argparse names to Streamlink session option key names:
https://github.com/streamlink/streamlink/blob/1d525705a0506327a5d518bfac04271cee74f875/src/streamlink_cli/argparser.py#L1229-L1285

> is it possible to set mpv `--volume` through these options?

No, this is completely unrelated. The Streamlink session is part of the `streamlink` package. Anything player- or output-related is done via `streamlink_cli`, which has streamlink as a dependency, but not the other way around (except for the `__main__` module). If you want to launch a player from your own python module, then you'll have to implement this yourself. Streamlink does only provide the various `Stream` and `StreamIO` interfaces.

**Reply by NiceShyGuy** [reply]
Thank You.
